src/utils.jl

```diff
@@ -77,6 +77,52 @@ function sideof(p::Point{2,T}, c::Chain{2,T}) where {T}
   ifelse(isapprox(w, zero(T), atol=atol(T)), :OUTSIDE, :INSIDE)
 end
 
+"""
+    sideof(point, mesh)
+
+Determines whether a `point` is inside, outside or on the surface of a `mesh`.
+Possible results are `:INSIDE`, `:OUTSIDE`, or `:ON`.
+
+### Notes
+
+Uses a ray-casting algorithm.
+"""
+function sideof(point::Point{3,T}, mesh::Mesh{3,T}) where {T}
+  if !(eltype(mesh) <: Triangle)
+    throw(ArgumentError(
+          "This function only works for surface meshes with triangles as elements."))
+  end
```

Comment:
Last question. In the case we have a polygon mesh with quadrangles or other types of polygons we can simply call `simplexify(mesh)` to obtain a triangle mesh before running the algorithm. So maybe we should check that the parametric dimension of the mesh is 2 first, and if it is only made of triangles we proceed, otherwise we simplexify and continue. What do you think?

```julia
@assert paramdim(mesh) == 2 "sideof only defined for surface meshes"

(eltype(mesh) <: Triangle) || return sideof(point, simplexify(mesh))
```

Reply:
Yeah that looks good!